app/views/devise/sessions/new.html.erb

```diff
@@ -0,0 +1,51 @@
+<div id="loginbox" style="margin-top:50px;" class="mainbox col-md-6 col-md-offset-3 col-sm-8 col-sm-offset-2">
+  <div class="panel panel-info" >
+    <div class="panel-heading" >
+      <div class="panel-title"><h3>Sign In</h3></div>
+    </div>
+
+    <div style="padding-top:30px" class="panel-body" >
+      <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+
+        <%= render "devise/shared/error_messages", resource: resource %>
```

Comment:
question: Why does the `error_messages` file have a `_` prefix and why is it not required when referencing it here?

Reply:
Files with `_` prefix are called partials in rails. Partials can be reused in different files. when we require it we don't have to prefix it.